plugins/chat/assets/javascripts/discourse/components/chat/thread/header.js

```diff
@@ -22,9 +26,49 @@ export default class ChatThreadHeader extends Component {
     );
   }
 
+  get threadNotificationLevel() {
+    return this.membership?.notificationLevel || NotificationLevels.REGULAR;
+  }
+
+  get membership() {
+    return this.args.thread.currentUserMembership;
+  }
+
   @action
   openThreadSettings() {
     const controller = showModal("chat-thread-settings-modal");
     controller.set("thread", this.args.thread);
   }
+
+  @action
+  updateThreadNotificationLevel(val) {
+    let originalVal;
+
+    if (this.membership) {
+      originalVal = this.membership.notificationLevel;
```

Comment:
With this and the `.catch` I am trying more to do the "optimistic UI updates" we have talked about where we don't wait for the server response to update the UI.

Reply:
I guess there's a point to be made where we should encapsulate these complex bits of mutations outside of components which are mostly here to display data but that's already nice like this.

I might give a more descriptive name to `val` and `originalVal`, like `newNotificationLevel`, `currentNotificationLevel` but up to you.

Reply:
Yep makes sense, will fix the naming :+1: I did think about that encapsulation too, but I think for now since there is only one place you can change this it's ok to just have it in the component.